src/compas/artists/artist.py

```diff
@@ -27,35 +27,73 @@ def redraw():
 
 @pluggable(category="factories", selector="collect_all")
 def register_artists():
+    """Registers artists available in the current context."""
     raise NotImplementedError
 
 
-def identify_context():
+def is_viewer_open():
+    """Returns True if an instance of the compas_view2 App is available.
+
+    Returns
+    -------
+    bool
+
+    """
+    # TODO: implement [without introducing compas_view2 as a dependency..?]
+    return False
+
+
+def is_plotter_open():
+    """Returns True if an instance of the Plotter is available.
+
+    Returns
+    -------
+    bool
+
+    """
+    # TODO: implement
+    return False
+
+
+def _choose_context():
+    """Chooses an appropriate context depending on available contexts and open instances. with the following priority:
+    1. Viewer
+    2. Plotter
+    3. Rhino / GH
+    4. Other - randome choice from Artist.ITEM_ARTISTS.values()
+
+    Returns
+    -------
+    str
+        one-of: ["Viewer", "Plotter", "Grasshopper", "Rhino"] or another as detected by the register_artists plugin.
+
+    """
+    if is_viewer_open():
+        return "Viewer"
+    if is_plotter_open():
+        return "Plotter"
     if compas.is_grasshopper():
         return "Grasshopper"
     if compas.is_rhino():
         return "Rhino"
     if compas.is_blender():
         return "Blender"
-    return None
+    other_contexts = [v for v in Artist.ITEM_ARTIST.keys() if v not in Artist.KNOWN_CONTEXTS]
+    if other_contexts:
+        return other_contexts[0]
```

Comment:
i assume this is to accommodate the context you are working on now? sounds like a new problem in the making as soon as there is more than 1. a bit like with the plugins...